redbox-core/redbox/graph/README.md

```diff
@@ -0,0 +1,45 @@
+# Redbox and LangGraph
+
+The core functionality of Redbox is fundamentally a graph. The user submits a request, and we run steps based on that request until we've finished our result. It's a bit like a factory production line: request in, sequential operations, result out.
+
+Graphs get complicated. We therefore present the following constraints for developing the graph, which we believe will mean the graph never gets too much more complicated to understand than it already is.
+
+> [!IMPORTANT]
+> [Low-Level LangGraph Concepts](https://langchain-ai.github.io/langgraph/concepts/low_level/) is essential reading for developing this part of Redbox.
+
+## Design conventions
+
+Draw your change before you do it in code.
+
+## Node conventions
+
+The graph of Redbox contains three types of node: robots doing jobs on the production line.
+
+* Processes, which affect the state
+* Decisions, which have conditional out-edges
+* Sends, which have conditional send out-edges
+
+Naming conventions should be honoured:
+
+* `p_`rocesses should be named for what they do: `p_summarise`, `p_retrieve_documents`
```

Comment:
tbh i would just type these out long-form and not use abbreviations (just a comment, no need to do anything)